cli/pkg/kctrl/local/sources/git.go

```diff
@@ -11,62 +11,61 @@ import (
 	vendirconf "github.com/vmware-tanzu/carvel-vendir/pkg/vendir/config"
 )
 
-type GitStep struct {
-	ui           cmdcore.AuthoringUI
-	vendirConfig vendirconf.Config
+type SourceConfiguration interface {
+	Configure() error
 }
 
-func NewGitStep(ui cmdcore.AuthoringUI, vendirConfig vendirconf.Config) *GitStep {
-	return &GitStep{
-		ui:           ui,
-		vendirConfig: vendirConfig,
-	}
+type GitConfiguration struct {
+	ui           cmdcore.AuthoringUI
+	vendirConfig *VendirConfig
 }
 
-func (g *GitStep) PreInteract() error { return nil }
+var _ SourceConfiguration = &GitConfiguration{}
 
-func (g *GitStep) Interact() error {
-	contents := g.vendirConfig.Directories[0].Contents
+func NewGitConfiguration(ui cmdcore.AuthoringUI, vendirConfig *VendirConfig) *GitConfiguration {
+	return &GitConfiguration{ui: ui, vendirConfig: vendirConfig}
+}
+
+func (g *GitConfiguration) Configure() error {
+	contents := g.vendirConfig.Contents()
 	if contents == nil {
-		err := g.initializeContentWithGit()
+		err := g.initializeContentWithGit(contents)
```

Comment:
I agree with @rohitagg2020, passing contents seems to be creating a lot of noise here.

Reply:
We will be replacing `contents[0]` with `vendirConfig.Contents()[0]` in all the functions if we want to do that, true for all of these cases
Happy to do it if y'all think I should

Reply:
Maybe we can do the `contents := g.vendirConfig...` in the functions where we are using `vendirConfig.Contents()[0]` more frequently?